x-pack/plugins/rule_registry/server/alert_data_client/tests/get.test.ts

```diff
@@ -151,12 +173,12 @@ describe('get()', () => {
         },
       })
     );
-    await alertsClient.get({ id: '1', index: '.alerts-observability-apm' });
+    await alertsClient.get({ id: 'NoxgpHkBqbdrfX07MqXV', index: '.alerts-observability-apm' });
 
     expect(auditLogger.log).toHaveBeenCalledWith({
       error: undefined,
-      event: { action: 'alert_get', category: ['database'], outcome: 'success', type: ['access'] },
-      message: 'User has accessed alert [id=1]',
+      event: { action: 'alert_get', category: ['database'], outcome: 'unknown', type: ['access'] },
+      message: 'User is accessing alert [id=NoxgpHkBqbdrfX07MqXV]',
```

Comment:
You have a test case for a `get()` that results in a successful audit event; what about an audit error when the user is unauthorized?

Reply:
Yes I think I actually removed the test case because with the `get` we were relying on a search executed with the authorization filter from the alerting framework to only give us back the data we were authorized to view, so there would be no authorization failure because we never attempted to access it in the first place.

I changed this so we are not solely rely on the authorization filter. We also call into `ensureAuthorized` on each search result (+ utilizing the authorization filter) so adding in an audit failure test makes sense again.

Fixed here: 36838ab2e10ba9b2640665345758373075e0fa44